docs/components/concepts/process-instance-migration.md

```diff
@@ -176,43 +286,56 @@ You can use [process instance modification](./process-instance-modification.md)
 This results in new keys for the service task as well as the job.
 :::
 
-## Dealing with attached boundary events
+## Internal Execution
 
-You can migrate active elements with boundary events attached.
-You decide what happens to the associated event subscription through the mapping instructions for the boundary events:
+In the following example, we will explain the internal execution steps of process instance migration.
+It is recommended that you read these steps to fully understand the power of process instance migration.
 
-- If a boundary event is mapped, the associated subscription is migrated. Please note, this case is not yet supported.
-- If a boundary event in the source process is not mapped, then the associated subscription is closed during migration.
-- If a boundary event of the target process is not the target of a mapping instruction, then a new subscription is opened during migration.
+Migration of a process instance consists of the following steps:
 
-Let's consider a process instance awaiting at a service task `A`.
+- Validation
+- Migration of mapped active elements
+- Unsubscribing from catch events
+- Subscribing to catch events
+- Migrating catch events subscriptions
```

Comment:
🔧 The steps are actually more involved and we don't migrate all active elements before dealing with the catch events. More correct would be the following.
```suggestion
- Validation
- Migration of process instance
- Migration of global variables
- For each active element
  - Migration of element instance
  - Migration of associated job
  - Migration of associated incident
  - Migration of associated user task
  - Migration of local variables
  - Unsubscribing from catch events
  - Subscribing to catch events
  - Migrating catch events subscriptions
```

💭 However, we should consider whether we want to go into so much detail about the implementation, as it means it becomes harder the change if we have such a need in the future. What does the reader benefit from this knowledge? If we see a need to document this, then I'd suggest a more ambiguous description to allow more flexibility in the future:

```
- Validation
- Migration of process instance, and global variables
- Migration of each active element (including associated jobs, incidents, user tasks, local variables, and event subscriptions)
```

Reply:
> What does the reader benefit from this knowledge?

Good question. Using migration tool can be complicated especially for bigger process definitions with a lot of active elements and we have a bunch of rejection messages. Explaining the order of execution would be useful to reason about these rejections and their orders. I also agree, we can further simplify this by your later suggestion.

❌ to myself. While overviewing the new page before merge. I realised, I indicated the execution order the other way around. It is actually breadth first. I will fix that one as well.

Reply:
I chose to opt-out mentioning user tasks because it is just an implementation detail of user tasks.